kedro/extras/datasets/pandas/json_dataset.py

```diff
@@ -22,7 +22,7 @@
 logger = logging.getLogger(__name__)
 
 
-class JSONDataSet(AbstractVersionedDataSet):
+class JSONDataSet(AbstractVersionedDataSet[pd.DataFrame, Any]):
```

Comment:
```suggestion
class JSONDataSet(AbstractVersionedDataSet[pd.DataFrame, pd.DataFrame]):
```
And change the type signature for `_load` 🙂 It does't look right at the moment.